grp/classic.gd

```diff
@@ -993,7 +993,125 @@ DeclareSynonym( "PSp", PSP );
 
 #############################################################################
 ##
-#O  ProjectiveOmegaCons( <filt>, <e>, <d>, <q> )
+#F  DECLARE_PROJECTIVE_ORTHOGONAL_GROUPS_OPERATION( ... )
+##
+BindGlobal("DECLARE_PROJECTIVE_ORTHOGONAL_GROUPS_OPERATION",
+  # ( <name>, <abbreviation>, <sizefunc-or-fail> )
+  function( nam, abbr, szf )
+    local pnam,cons,opr;
+
+    opr:= VALUE_GLOBAL( nam );
+    pnam:= Concatenation( "Projective", nam );
+    cons:= NewConstructor( Concatenation( pnam, "Cons" ),
+                           [ IsGroup, IsInt, IsInt, IsInt ] );
+    BindGlobal( Concatenation( pnam, "Cons" ), cons );
+    BindGlobal( pnam, function( arg )
+      if Length( arg ) = 2 then
+        return cons( IsPermGroup, 0, arg[1], arg[2] );
+      elif Length( arg ) = 3 and ForAll( arg, IsInt ) then
+        return cons( IsPermGroup, arg[1], arg[2], arg[3] );
+      elif IsOperation( arg[1] ) then
+        if Length( arg ) = 3 then
+          return cons( arg[1], 0, arg[2], arg[3] );
+        elif Length( arg ) = 4 then
+          return cons( arg[1], arg[2], arg[3], arg[4] );
+        fi;
+      fi;
+      Error( "usage: ", pnam, "( [<filter>, ][<e>, ]<d>, <q> )" );
+    end );
+
+  DeclareSynonym( Concatenation( "P", abbr ), VALUE_GLOBAL( pnam ) );
+
+  # Install a method to get the permutation action on lines.
+  InstallMethod( cons, "action on lines",
+    [ IsPermGroup, IsInt, IsPosInt, IsPosInt ],
+    function( filter, e, n, q )
+    local g, p;
+
+    g:= opr( IsMatrixGroup, e, n, q );
+    p:= ProjectiveActionOnFullSpace( g, GF( q ), n );
+    if szf <> fail then
+      SetSize( p, szf( e, n, q, g ) );
+    fi;
+
+    return p;
+    end );
+end );
+
+
+#############################################################################
+##
+#F  ProjectiveGeneralOrthogonalGroup( [<filt>, ][<e>, ]<d>, <q> )
+#F  PGO( [<filt>, ][<e>, ]<d>, <q> )
+##
+##  <#GAPDoc Label="ProjectiveGeneralOrthogonalGroup">
+##  <ManSection>
+##  <Func Name="ProjectiveGeneralOrthogonalGroup" Arg='[filt, ][e, ]d, q'/>
+##  <Func Name="PGO" Arg='[filt, ][e, ]d, q'/>
+##
+##  <Description>
+##  constructs a group isomorphic to the projective group
+##  PGO( <A>e</A>, <A>d</A>, <A>q</A> )
+##  of GO( <A>e</A>, <A>d</A>, <A>q</A> ),
+##  modulo the centre
```

Comment:
A nitpick (which really is about existing documentation, which you just covered, so I don't really expect it to be addressed here): Formally, the definition is not to take the quotient modulo the center, but rather modulo scalar matrices (achieved implicitly by using a projective action on the vector). The two are related by not completely trivial mathematics. I think for a user who is not yet very experienced, this can be confusing (source: myself, many years ago ;-) -- so maybe I am just projecting here (pun not intended) and nobody else ever got confused).

Anyway, I guess what I am saying is that it might be better if we described in a separate section / paragraph what we mean by projective group (modulo scalars, equiv: acting on lines) and that this means we take a central factor, and that for many important cases ("full" groups) this in fact means factoring out the whole center. And then we can just reference that.

Thoughts? If people are not adverse to this, I could give it a go some of these days (I'd probably for now just submit an issue to not forget it...)

Reply:
Such a section on projective actions would be useful. It could refer to the FinInG package. (I had expected that also the recog package contains something interesting in this respect, but I did not find this at first sight.) In the other direction, the functions `NormedVectors`, `OnLines`, `ProjectiveActionOnFullSpace`, `ProjectiveOrder` could refer to the new section.